src/ca.cpp

```diff
@@ -98,7 +98,13 @@ X509Certificate CertificateAuthority::_signCSR(const CertificateSigningRequest &
     //Sanity check: certificate must be verifiable now
     try {
         auto cert = X509Certificate{std::move(newCertificate)};
-        cert.verify({_rootCert}, {});
+        X509Certificate::VerificationContext ctx;
+        ctx.addTrustedCertificates({_rootCert})
+            .addIntermediateCertificates({})
+            .setVerificationCheckTime(cert.getNotBeforeAsn1());
+        cert.verify(ctx);
+        ctx.setVerificationCheckTime(cert.getNotAfterAsn1() - Asn1Time::Seconds(1));
```

Comment:
If we need this, please add a comment due to the asymmetrie with NotBefore. However, I'd also think that it is not necessary.

Reply:
I added a comment, that's indeed how OpenSSL checks it